// @pnpm/npm-conf/lib/types.js
// Generated with `lib/make.js`
'use strict';
const path = require('path');
const Stream = require('stream').Stream;
const url = require('url');

const Umask = () => {};
const getLocalAddresses = () => [];
const semver = () => {};

exports.types = {
	access: [null, 'restricted', 'public'],
	'allow-same-version': Boolean,
	'always-auth': Boolean,
	also: [null, 'dev', 'development'],
	audit: Boolean,
	'auth-type': ['legacy', 'sso', 'saml', 'oauth'],
	'bin-links': Boolean,
	browser: [null, String],
	ca: [null, String, Array],
	cafile: path,
	cache: path,
	'cache-lock-stale': Number,
	'cache-lock-retries': Number,
	'cache-lock-wait': Number,
	'cache-max': Number,
	'cache-min': Number,
	cert: [null, String],
	cidr: [null, String, Array],
	color: ['always', Boolean],
	depth: Number,
	description: Boolean,
	dev: Boolean,
	'dry-run': Boolean,
	editor: String,
	'engine-strict': Boolean,
	force: Boolean,
	'fetch-retries': Number,
	'fetch-retry-factor': Number,
	'fetch-retry-mintimeout': Number,
	'fetch-retry-maxtimeout': Number,
	git: String,
	'git-tag-version': Boolean,
	'commit-hooks': Boolean,
	global: Boolean,
	globalconfig: path,
	'global-style': Boolean,
	group: [Number, String],
	'https-proxy': [null, url],
	'user-agent': String,
	'ham-it-up': Boolean,
	'heading': String,
	'if-present': Boolean,
	'ignore-prepublish': Boolean,
	'ignore-scripts': Boolean,
	'init-module': path,
	'init-author-name': String,
	'init-author-email': String,
	'init-author-url': ['', url],
	'init-license': String,
	'init-version': semver,
	json: Boolean,
	key: [null, String],
	'legacy-bundling': Boolean,
